platform/CircularBuffer.h

```diff
@@ -18,6 +18,20 @@
 
 #include "platform/mbed_critical.h"
 
+/* Detect if CounterType of the Circular buffer is of unsigned type. */
```

Comment:
Could you put the traits in a namespace named *internal* within the mbed namespace.

Reply:
Fixed as suggested.